--- a/lib/systems/parse.nix
+++ b/lib/systems/parse.nix
@@ -634,6 +634,10 @@
         execFormat = unknown;
         families = { };
       };
+      cosmo = {
+        execFormat = elf;
+        families = { };
+      };
     }
     // {
       # aliases
--- a/lib/systems/inspect.nix
+++ b/lib/systems/inspect.nix
@@ -362,6 +362,9 @@
     isNone = {
       kernel = kernels.none;
     };
+    isCosmo = {
+      kernel = kernels.cosmo;
+    };
 
     isAndroid = [
       { abi = abis.android; }
--- a/lib/systems/examples.nix
+++ b/lib/systems/examples.nix
@@ -211,6 +211,11 @@
     libc = "newlib";
   };
 
+  cosmo = {
+    config = "x86_64-unknown-cosmo-gnu";
+    libc = null;
+  };
+
   rx-embedded = {
     config = "rx-none-elf";
     libc = "newlib";
